onnxruntime/core/framework/instant_op.cc

```diff
@@ -0,0 +1,273 @@
+                                   output_count,
+                                   allocator,
+                                   ctx->GetOperatorThreadPool(),
+                                   ctx->Logger());
+  auto kernel = reinterpret_cast<const OpKernel*>(ort_op);
+  return kernel->Compute(&instant_ctx);
+}
+
+}  // namespace instant
+}  // namespace onnxruntime
+
+ORT_API_STATUS_IMPL(OrtApis::CreateAttribute,
+                    _In_ const char* name,
+                    _In_ const void* data,
+                    _In_ int len,
+                    _In_ ONNXTensorElementDataType type,
+                    _In_opt_ int is_array,
+                    _Outptr_ OrtOpAttr* op_attr) {
+  API_IMPL_BEGIN
+  if (!name || !data || !len || !op_attr) {
+    return CreateStatus(ORT_INVALID_ARGUMENT, "Invalid argument for CreateAttribute.");
+  }
+  auto status = onnxruntime::instant::CreateAttribute(name, data, len, type, is_array, op_attr);
+  if (status.IsOK()) {
+    return nullptr;
+  } else {
+    return CreateStatus(static_cast<OrtErrorCode>(status.Code()), "Failed to create attribute.");
+  }
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtApis::ReleaseAttribute, _Inout_ OrtOpAttr* op_attr) {
+  API_IMPL_BEGIN
+  if (op_attr && *op_attr) {
+    delete reinterpret_cast<ONNX_NAMESPACE::AttributeProto*>(*op_attr);
+    *op_attr = nullptr;
+  }
+  return nullptr;
+  API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(OrtApis::CreateOperator,
+                    _In_ const OrtKernelInfo* info,
+                    _In_ const char* op_name,
+                    _In_ const char* domain,
+                    _In_ int version,
+                    _In_opt_ const char** type_constraint_names,
+                    _In_opt_ const ONNXTensorElementDataType* type_constraint_values,
+                    _In_opt_ int type_constraint_count,
+                    _In_opt_ const OrtOpAttr* attr_values,
+                    _In_opt_ int attr_count,
+                    _Outptr_ OrtOp* ort_op) {
+  API_IMPL_BEGIN
+  if (!info || !op_name || !domain || !ort_op) {
+    return CreateStatus(ORT_INVALID_ARGUMENT, "Invalid argument for CreateOperator.");
+  }
+  auto status = onnxruntime::instant::CreateOperator(info,
+                                                     op_name,
+                                                     domain,
+                                                     version,
+                                                     type_constraint_names,
+                                                     type_constraint_values,
+                                                     type_constraint_count,
+                                                     attr_values,
+                                                     attr_count,
+                                                     ort_op);
+  if (status.IsOK()) {
+    return nullptr;
+  } else {
+    return CreateStatus(static_cast<OrtErrorCode>(status.Code()), "Failed to create operator.");
```

Comment:
Maybe append the error message from the internal status to this ?

Reply:
:heavy_check_mark:

Reply:
Done.